### PR TITLE
[Docs] Revise Vizro-AI docs to align with style guide

### PR DESCRIPTION
**Pull request by stichbury**
## Description
Now ready for review. You can see the built docs here: https://vizro--381.org.readthedocs.build/projects/vizro-ai/en/381/

I have made changes that should be familiar given the set I've made on `vizro-core` to follow the style guide. For example, I've change capitalisation in headers to sentence case.

I also rationalised the installation guide to align it more closely to the one for `vizro-core` and I converted a page to FAQs since it was better titled to answer common queries. The LLM API key content moves into the installation guide since, for now at least, it's covering just OpenAI and doesn't need to keep redirecting the reader around the docs to different places. I think getting the `.env` file set up is probably the most confusing thing for novice users who are unfamiliar with API keys and environment settings...we need to think about about a video at some point.

I moved some text around on the index page to remove the most "promotional" copy (that can go into the README if we need it) and keep it short and useful. The box navigation is now above the fold.

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/docs/pages/user-guides/install.md [hunk 1]
```diff
@@ -1,68 +1,167 @@
...
+## Install Vizro
+
+To install Vizro-AI, use [`pip`](https://pip.pypa.io/en/stable/) in your terminal window:
 
 ```bash
 pip install vizro_ai
 ```
 
-While you can execute code from the tutorials and user guides using a Python script, using a Jupyter notebook is often considered more convenient. You can install `jupyter` with the following command:
+## Use Vizro-AI inside Anaconda Navigator
```
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/install.md:129]: This PR is still in draft so I'm not after full review right now, but I have a question please @antonymilne or @maxschulz-COL.

Do we anticipate that anyone will want to use Vizro-AI within Anaconda Navigator.

I would prefer to remove this section if not because we've asked them to set up access to a LLM, which requires some technical know-how, and it seems to me if they can do that, they're not likely to be upset about using a terminal, so won't want to use Navigator to avoid terminal usage.

I'm happy to retain it in Vizro but wondered about dropping this section from these docs.

WDYT?

**Review by nadijagraca — APPROVED** [review]
Really good job Jo. I love the new docs. 🚀

File: vizro-ai/docs/index.md [hunk 2]
```diff
@@ -1,47 +1,32 @@
 # Vizro-AI
 
-Vizro-AI is a tool designed for generating data visualizations. It serves as an extension to Vizro, leveraging natural language capabilities to empower users in creating charts effortlessly.
+Vizro-AI extends [Vizro](https://vizro.readthedocs.io) with a user-friendly interface that uses natural language capabilities to help you create interactive charts effortlessly.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/index.md:3]: What do we consider as user-friendly interface here? Because Vizro-AI doesn't provide a user-friendly interface out of the box, does it? The user actually has to create it e.g. the chatbot was a use case we had to configure, but the vizro-ai package itself doesn't provide any UI interface. Or does this not refer to a UI interface?

File: vizro-ai/docs/index.md [hunk 3]
```diff
@@ -64,11 +49,19 @@ Vizro-AI leverages the power of [Plotly](https://plotly.com/python/) to produce
...
 </div>
 </div>
+
+!!! notice "Notice"
+    Please review this [disclaimer](pages/explanation/disclaimer.md)
+    before using the `vizro-ai` package.
+
+    Users must connect to Large Language Models (LLMs) to use Vizro-AI.
+    Please review our [guidelines on the use of LLMs](pages/explanation/safety-in-vizro-ai.md)
+    and the required [safeguarding for dynamic code evaluation](pages/explanation/safeguard.md).
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/index.md:67]: Not sure what our common syntax is but here we capitalise `Large Language Models` while in other sections we have everything in lower-case. So whatever you think is the better choice, I would stick with that 👍 

```suggestion
    Users must connect to large language models (LLMs) to use Vizro-AI.
    Please review our [guidelines on the use of LLMs](pages/explanation/safety-in-vizro-ai.md)
    and the required [safeguarding for dynamic code evaluation](pages/explanation/safeguard.md).
```

File: vizro-ai/docs/pages/explanation/safety-in-vizro-ai.md [hunk 4]
```diff
@@ -62,4 +61,4 @@ Users must exercise caution when executing code generated by or influenced by AI
 - Always be aware that malicious code execution cannot be mitigated with 100% certainty
 - Always review and understand the selected model before connecting with `vizro_ai`
 
-If you would like to learn more about our efforts in safeguarding code execution, please refer to [Safeguard Execution of Dynamic Code](safeguard.md) for more information.
+If you would like to learn more about our efforts in safeguarding code execution, refer to [Safeguard Execution of Dynamic Code](safeguard.md) for more information.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/explanation/safety-in-vizro-ai.md:64]: I am always confused what we capitalise and what we keep lower-case e.g. below I would have replicated how the title of that section is actually displayed so only capitalising the first letter? 🤔 

```suggestion
If you would like to learn more about our efforts in safeguarding code execution, refer to [Safeguard execution of dynamic code](safeguard.md) for more information.
```

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md [hunk 5]
```diff
@@ -1,29 +1,23 @@
 # Explore Vizro-AI
-This tutorial serves as an exploration of the extended applications offered by Vizro-AI beyond the initial [quickstart guide](../tutorials/quickstart.md).
-
-By the end of this tutorial, you will have gained an understanding of different language options and leveraging Vizro-AI to enhance the formatting your visualizations.
-
-## Let's get started!
-### 1. Install Vizro-AI and get ready to run your code
-Before proceeding, ensure the installation of the `vizro_ai` package by following the steps outlined in the [installation guide](../user-guides/install.md). Once installed, you can execute your code either by pasting it into a Jupyter notebook cell or running it from a Python script.
+This tutorial serves as an exploration of Vizro-AI beyond the initial [quickstart guide](../tutorials/quickstart.md). It explains the different language options and shows how to use Vizro-AI to enhance the formatting your visualizations.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:2]:
```suggestion
This tutorial serves as an exploration of Vizro-AI beyond the initial [quickstart guide](../tutorials/quickstart.md). It explains the different language options and shows how to use Vizro-AI to enhance the formatting of your visualizations.
```

**Comment by maxschulz-COL** [comment]
I am being a pain, I cannot approve this as you didn't click the button 😂  @Joseph-Perkins

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md [hunk 6]
```diff
@@ -1,29 +1,23 @@
...
 
+## Install Vizro-AI and get ready to run your code
+Before proceeding, you'll need to install Vizro-AI by following the steps in the [installation guide](../user-guides/install.md). To execute code either paste it into a Jupyter Notebook or run it from a Python script, or see the [guide on how to run Vizro-AI](../user-guides/run-vizro-ai.md) for more detail.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:5]: Quick question as I also acknowledge that we have been quite inconsistent in the past - what should we include in a link? e.g.

- [user guide] on how to run Vizro-AI
- [guide on how to run Vizro-AI]
- guide on [how to run Vizro-AI]

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md [hunk 7]
```diff
@@ -42,6 +36,7 @@ Vizro-AI is versatile, supporting prompts and chart visualizations in multiple l
 Subsequently, we'll switch to German and prompt the visualization of life expectancy in the United States over time, comparing it to the global life expectancy trend. For this example, we'll include `explain=True` to obtain comprehensive insights into both the data and the generated code.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:36]: Where is the Chinese example coming from? 😄 Have we double-checked with someone that actually speaks it? haha

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md [hunk 8]
```diff
@@ -57,12 +52,13 @@ Subsequently, we'll switch to German and prompt the visualization of life expect
 
     [GermanChart]: ../../assets/tutorials/chart/GermanExample.png
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:53]: @maxschulz-COL - "kreiere eine deutsche Visualisierung" - is this supposed to be like this? 😅  What does that change? Isn't that part redundant because it will create a visualisation in German anyways if the prompt is in German?

File: vizro-ai/docs/pages/tutorials/quickstart.md [hunk 9]
```diff
@@ -1,55 +1,74 @@
...
+
 
-### 3. Large Language Model (LLM) API KEY
+### 3. Create your first chart using Vizro-AI
 
-A prerequisite to use Vizro-AI is access to one of the supported LLMs. Refer to the [user guide](../user-guides/api-setup.md) on how to set up the API.
+Let's create a chart to illustrate the GDP of various continents while including a reference line for the average. We give Vizro-AI the English language instruction "*describe the composition of GDP in continent and color by continent, and add a horizontal line for avg GDP*".
 
-### 4. Ask your first question using Vizro-AI
+Let's go through the code step-by-step. First, we create `pandas` DataFrame using the gapminder data from `plotly express`:
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/tutorials/quickstart.md:46]:
```suggestion
Let's go through the code step-by-step. First, we create a `pandas` DataFrame using the gapminder data from `plotly express`:
```

File: vizro-ai/mkdocs.yml [hunk 10]
```diff
@@ -3,18 +3,16 @@ site_url: https://vizro-ai.readthedocs.io/en/latest/
 nav:
   - Vizro-AI: index.md
   - Get started:
-      - Quickstart: pages/tutorials/quickstart.md
+      - Install Vizro-AI: pages/user-guides/install.md
+      - A first dashboard: pages/tutorials/quickstart.md
```
Comment (maxschulz-COL) [inline vizro-ai/mkdocs.yml:7]: One does not create a dashboard

File: vizro-ai/docs/pages/user-guides/install.md [hunk 11]
```diff
@@ -1,68 +1,156 @@
...
-- **Virtual environment**: We recommend creating a new virtual environment for each new Vizro-AI project you work on to
-  isolate the Python dependencies from those of other projects. To learn more, please see the following references about [python virtual environments](https://realpython.com/python-virtual-environments-a-primer/), [conda virtual environments](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#starting-conda) or [watch an explainer video about them](https://youtu.be/YKfAwIItO7M).
-- **OpenAI API Key**: Using Vizro-AI requires an API key, which you can get by creating an account [here](https://platform.openai.com/account/api-keys). (Users are recommended to review the third party API key section of the [disclaimer](../explanation/disclaimer.md) documentation.
-). Once you have your API key, the next step is to set your environment variable by:
-```bash
-export OPENAI_API_KEY="..."
+### Python
+Vizro-AI supports macOS, Linux, and Windows. It works with Python 3.8 and later. You can specify the version of Python to use with Vizro-AI when you set up a virtual environment.
```
Comment (huong-li-nguyen) [inline vizro-ai/docs/pages/user-guides/install.md:8]: As far as I remember, Vizro-AI doesn't work with 3.8? We also say 3.9 and later in the pypi: https://pypi.org/project/vizro-ai/ but please double-check with @maxschulz-COL 

```suggestion
Vizro-AI supports macOS, Linux, and Windows. It works with Python 3.9 and later. You can specify the version of Python to use with Vizro-AI when you set up a virtual environment.
```

File: vizro-ai/docs/pages/tutorials/quickstart.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/quickstart.md:1]: Should this not maybe the only guide under quickstart? This is the tutorial section, and it is nice to have on E2E example here

**Review by huong-li-nguyen — COMMENTED** [review]
LGTM in general 👍 Have a few comments and question!

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:1]: Subsequently to my other comment - should we kill this page and transfer the secions here to individual "how-to" guides?

File: vizro-ai/mkdocs.yml [hunk 12]
```diff
@@ -3,18 +3,16 @@ site_url: https://vizro-ai.readthedocs.io/en/latest/
...
       - Explore Vizro-AI: pages/tutorials/explore-vizro-ai.md
-  - User Guides:
-      - Fundamentals:
-          - Installation: pages/user-guides/install.md
```
Comment (maxschulz-COL) [inline vizro-ai/mkdocs.yml:10]: Where has the installation guide gone e.g.? I think this belongs to a "how to" guide no?

**Review by maxschulz-COL — COMMENTED** [review]
Hey Jo, I think we can still work a little bit on shuffling the sections. I think also some of the pages have gotten lost 🤔 ?

File: vizro-ai/docs/pages/user-guides/install.md (hunk 1 above)
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/user-guides/install.md:118]: I agree! Let's remove that section!

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md (hunk 7 above)
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:36]: Lingyi wrote it!

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md (hunk 8 above)
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:53]: Haha good question - @chiara-sophie was there a reason for this specific prompt?

File: vizro-ai/docs/pages/user-guides/install.md (hunk 11 above)
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/user-guides/install.md:8]: Yes, let's stick with `3.9` and higher

File: vizro-ai/docs/index.md (hunk 2 above)
Comment (stichbury) [reply vizro-ai/docs/index.md:3]: That's a good point! I've made an edit to revise this.

File: vizro-ai/docs/index.md (hunk 3 above)
Comment (stichbury) [reply vizro-ai/docs/index.md:67]: Thanks for the spot! I thought I'd caught all of those but I missed this one. Heading off to grep again. The rule is:

1. Capitalise in the acronym (so it's "LLM")
2. Don't capitalise the full name because it's not a proper noun (it's "large language models") UNLESS at the start of a sentence or header, then it's "Large language models"

File: vizro-ai/docs/pages/explanation/safety-in-vizro-ai.md (hunk 4 above)
Comment (stichbury) [reply vizro-ai/docs/pages/explanation/safety-in-vizro-ai.md:64]: You're right, but in fact that was all a bit of a mess, so I've fixed it in your suggested edit. Thanks! 🎯

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md (hunk 6 above)
Comment (stichbury) [reply vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:5]: It doesn't matter too much. It's better to have more detail for people using screen readers (if they end up with a set of links that say "user guide" it doesn't help them navigate the docs much). I'd go with the full "go on how to run..." as per my edit here, but the third option is also fine.

File: vizro-ai/docs/index.md (hunk 3 above)
Comment (huong-li-nguyen) [reply vizro-ai/docs/index.md:67]: Perfect, all clear now! 👍

File: vizro-ai/mkdocs.yml (hunk 10 above)
Comment (stichbury) [reply vizro-ai/mkdocs.yml:7]: Oops! Fixed to "A first example"

File: vizro-ai/mkdocs.yml (hunk 12 above)
Comment (stichbury) [reply vizro-ai/mkdocs.yml:10]: I've moved the page in the layout to the "Get started" section as I did for `vizro-core`. I agree that it is a how-to guide in that it's shows how to install, but I didn't like the user jumping from quickstart straight away into how-to, and decided to bring installation directly into the get started instructions (as, after all, they can't get started until they've done the install).

https://vizro--381.org.readthedocs.build/projects/vizro-ai/en/381/pages/user-guides/install/

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md
Comment (stichbury) [reply vizro-ai/docs/pages/tutorials/explore-vizro-ai.md:1]: Yes, I agree, let's do that!

File: vizro-ai/docs/pages/tutorials/quickstart.md
Comment (stichbury) [reply vizro-ai/docs/pages/tutorials/quickstart.md:1]: Agreed!

**Review by huong-li-nguyen — APPROVED** [review]
🚀

File: vizro-ai/docs/pages/user-guides/use-different-languages.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+        vizro_ai = VizroAI()
+        vizro_ai.plot(df, "Visualiere den Trend von der Lebenserwartung in USA über die Jahre im Vergleich zur Veränderung der weltweiten Lebenserwartung über die Jahre und kreiere eine deutsche Visualisierung", explain=True)
+        ```
+    === "Result"
+        [![GermanChart]][GermanChart]
+
+    [GermanChart]: ../../assets/tutorials/chart/GermanExample.png
+
+## How to create advanced charts and formatting
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/use-different-languages.md:40]: Is this still supposed to be in?

**Review by maxschulz-COL — CHANGES_REQUESTED** [review]
I think we have a duplicated section, but after that I think we are good to go! Really good overhaul, and it is already much more concise 💪 

One last idea: shall we move the FAQs to the Explanation section, then I think we are super clean and following the diataxis framework well!

File: vizro-ai/docs/pages/tutorials/explore-vizro-ai.md (hunk 8 above)
Comment (huong-li-nguyen) [reply vizro-ai/docs/pages/tutorials/explore-vizro-ai.md]: @maxschulz-COL - could you try running the example and see if we need it? Preferably I would cut that part from the docs as it sounds a bit weird :P Otherwise, I would rephrase to the suggested above?

File: vizro-ai/docs/pages/user-guides/use-different-languages.md (hunk 13 above)
Comment (stichbury) [reply vizro-ai/docs/pages/user-guides/use-different-languages.md]: Fixed!
